controllers/rating_controller.go

```diff
@@ -82,10 +82,23 @@ func (r *RatingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		}
 		return reconcile.Result{}, err
 	}
-
+	instanceDeepCopy := instance.DeepCopy()
 	done, err := r.ratingChecker(ctx, &instance)
 	if !done {
-		return reconcile.Result{Requeue: true}, err
+		cond := corev1alpha1.Condition{
+			Status:             v1.ConditionFalse,
+			LastTransitionTime: metav1.Now(),
+			Reason:             corev1alpha1.ReasonReconcileError,
+			Message:            "Rating disabled in component's repository",
+			Type:               corev1alpha1.TypeReady,
+		}
+		instanceDeepCopy.Status.ConditionedStatus = corev1alpha1.ConditionedStatus{
+			Conditions: []corev1alpha1.Condition{cond},
+		}
+		if err := r.Client.Status().Patch(ctx, instanceDeepCopy, client.MergeFrom(&instance)); err != nil {
```

Comment:
```go
err := r.Client.Status().Patch(ctx, instanceDeepCopy, client.MergeFrom(&instance))
return reconcile.Result{}, err
```

Reply:
Is there something wrong here?

Reply:
@zqq454224016 I think @0xff-dev means you can optimize your code:
 - return ths without error check